dotCMS/src/main/java/com/dotcms/contenttype/business/DotAssetAPIImpl.java

```diff
@@ -0,0 +1,146 @@
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<ContentType> findDotAssetContentType(final String mimeType,
+                                                          final Map<String, ContentType>... mimeTypeMappingArray) {
+
+        for (final Map<String, ContentType> mimeTypeContentTypeMap : mimeTypeMappingArray) {
+
+            if (null != mimeTypeContentTypeMap) {
+                for (final Map.Entry<String, ContentType> entry : mimeTypeContentTypeMap.entrySet()) {
+
+                    if (MimeTypeUtils.match(entry.getKey(), mimeType)) {
+
+                        return Optional.ofNullable(entry.getValue());
+                    }
+                }
+            }
+        }
+
+        return Optional.empty();
+    }
+
+    private Map<String, ContentType> getMap(final Map<String, ContentType>[] mimeTypeMappingArray, final int index) {
+
+        if (null == mimeTypeMappingArray[index]) {
+
+            mimeTypeMappingArray[index] = new HashMap<>();
+        }
+
+        return mimeTypeMappingArray[index];
+    }
+
+    private ContentTypeMimeType mapToContentTypeMimeType(final ContentType contentType) {
+
+        final String systemHostId          = APILocator.systemHost().getIdentifier();
+        final String contentTypeHostId     = contentType.host();
+        final boolean isSystemHost         = null == contentTypeHostId || contentTypeHostId.equals(systemHostId);
+        final Map<String, Field>  fieldMap = contentType.fieldMap();
+        String [] mimeTypeFieldVariables   = new String [] { ALL_MIME_TYPE };
+        final List<FieldVariable> fieldVariables = fieldMap.get(DotAssetContentType.ASSET_FIELD_VAR).fieldVariables();
+        if (UtilMethods.isSet(fieldVariables)) {
+
+            final Optional<FieldVariable> fieldVariableOpt = fieldVariables.stream().filter(fieldVariable ->
+                    BinaryField.ALLOWED_FILE_TYPES.equalsIgnoreCase(fieldVariable.key())).findFirst();
+
+            if (fieldVariableOpt.isPresent() && UtilMethods.isSet(fieldVariableOpt.get().value())) {
+
+                mimeTypeFieldVariables = fieldVariableOpt.get().value().split(StringPool.COMMA);
+            }
+        }
+
+        return new ContentTypeMimeType(mimeTypeFieldVariables, contentType, isSystemHost);
+    }
+
+    private boolean isSystemHostOrCurrentHost (final ContentType contentType, final Host currentHost) {
+
+        final String contentTypeHostId = contentType.host();
+        final String systemHostId      = APILocator.systemHost().getIdentifier();
+        return null == contentTypeHostId ||
+                contentTypeHostId.equals(currentHost.getIdentifier()) || contentTypeHostId.equals(systemHostId);
+
+    }
+
+    private class ContentTypeMimeType {
+
+        private final boolean systemHost;
+        private final String [] mimeTypeFieldVariables;
+        private final ContentType contentType;
+
+        public ContentTypeMimeType(final String [] mimeTypeFieldVariables,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The user-supplied array 'mimeTypeFieldVariables' is stored directly.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028307)